tools/driver/p4c_src/main.py

```diff
@@ -45,25 +47,40 @@ def display_supported_targets(cfg):
         ret += str(target) + "\n"
     return ret
 
+
+JSON_input_flag = "--fromJson"  ### DRY
+
+
 def add_developer_options(parser):
     parser.add_argument("-T", dest="log_levels",
                         action="append", default=[],
-                        help="[Compiler debugging] Adjust logging level per file (see below)")
+                        help="[Compiler debugging] Adjust logging level"
+                        " per file (see below)")
     parser.add_argument("--top4", dest="passes",
                         action="append", default=[],
-                        help="[Compiler debugging] Dump the P4 representation after \
-                               passes whose name contains one of `passX' substrings. \
-                               When '-v' is used this will include the compiler IR.")
+                        help="[Compiler debugging] Dump the P4 representation"
+                             " after passes whose name contains one of 'passX'"
+                             " substrings.  When '-v' is used this will"
+                             " include the compiler IR.")
     parser.add_argument("--dump", dest="dump_dir", default=None,
-                        help="[Compiler debugging] Folder where P4 programs are dumped.")
+                        help="[Compiler debugging] Folder where P4 programs"
+                             " are dumped.")
     parser.add_argument("--toJson", dest="json", default=None,
                         help="Dump IR to JSON in the specified file.")
-    parser.add_argument("--fromJson", dest="json_source", default=None,
-                        help="Use IR from JSON representation dumped previously. \
-                        the compilation starts with reduced midEnd.")
+    parser.add_argument(JSON_input_flag, dest="json_source", default=None,
+                        help="Use IR from JSON representation dumped"
+                             " previously.  The compilation starts with a"
+                             " reduced midEnd.  Only allowed/recognized in"
+                             " developer builds.")
     parser.add_argument("--pp", dest="pretty_print", default=None,
                         help="Pretty-print the program in the specified file.")
 
+
+### a utility function for grammatical correctness [and DRY]
```

Comment:
In general I think it's better to use """ instead of # for this kind of documentation.

Reply:
Agreed; changed in my latest [local] Git.  Should be included in a batch of changes I intend to upload soon-ish.